src/Integration/Connection/ConnectionWorkflow.cs

```diff
@@ -183,6 +183,21 @@ private void OnProjectsChanged(ConnectionInformation connection, IEnumerable<Son
                 var projects = await this.host.SonarQubeService.GetAllProjectsAsync(connection.Organization?.Key,
                     cancellationToken);
 
+                // The SonarQube client will limit the number of returned projects to 10K (hard limit on SonarQube side)
+                // but will no longer fail when trying to retrieve more. In the case where the project we want to bind to
+                // is not in the list AND the binding was already done (the .slconfig was manually created) then we are
+                // going to pretend we managed to retrieve this project from the list.
+                if (string.IsNullOrEmpty(connection.Password.ToUnsecureString()) &&
```

Comment:
I don't understand why the check for an empty password is needed. What happens if the user has supplied credentials?

Reply:
From the different tests I have done, this is the best way I found to detect the case of a binding for a solution that was already bound. Basically when opening a solution which is already bound this password is empty for some reason (probably because we then use windows credential store).